datalad/distribution/dataset.py

```diff
@@ -126,11 +128,17 @@ def repo(self):
                             lgr.debug("Detected %s at %s", cls, self._path)
                             self._repo = cls(self._path, create=False, **kw)
                             break
-                        except (InvalidGitRepositoryError, NoSuchPathError, RuntimeError) as exc:
-                            lgr.debug("Oops -- guess on repo type was wrong?: %s", exc_str(exc))
+                        except (InvalidGitRepositoryError, NoSuchPathError) as exc:
+                            lgr.debug(
+                                "Oops -- guess on repo type was wrong?: %s",
+                                exc_str(exc))
                             pass
-                if self._repo is None:
-                    lgr.info("Failed to detect a valid repo at %s" % self.path)
+                        # version problems come as RuntimeError: DO NOT CATCH!
+            if self._repo is None:
+                # Often .repo is requested to 'sense' if anything is installed
```

Comment:
agree.  And that is why I lowered log msg from INFO to DEBUG, where it might still be useful.  E.g. that is how I detected that currently deducing "crumb datasets path" to an arbitrary path requires instantiation of datasets for all "neighboring" submodules and asking their `.repo` at some point (imho unnecessary slow-down):
```shell
$> datalad -l debug install openfmri/ds000113d/sub-10
2016-10-18 08:50:40,603 [DEBUG  ] Install passes into get 1 items 
2016-10-18 08:50:40,603 [DEBUG  ] Resolved targets to get: ['/home/yoh/datalad/openfmri/ds000113d/sub-10'] 
2016-10-18 08:50:40,603 [DEBUG  ] Found 0 existing dataset(s) to get content in and 1 unavailable paths 
2016-10-18 08:50:40,918 [DEBUG  ] Failed to detect a valid repo at /home/yoh/datalad/openfmri/ds000030 
2016-10-18 08:50:40,927 [DEBUG  ] Failed to detect a valid repo at /home/yoh/datalad/openfmri/ds000031 
2016-10-18 08:50:40,935 [DEBUG  ] Failed to detect a valid repo at /home/yoh/datalad/openfmri/ds000051 
2016-10-18 08:50:40,948 [DEBUG  ] Failed to detect a valid repo at /home/yoh/datalad/openfmri/ds000052 
2016-10-18 08:50:40,958 [DEBUG  ] Failed to detect a valid repo at /home/yoh/datalad/openfmri/ds000053 
2016-10-18 08:50:40,968 [DEBUG  ] Failed to detect a valid repo at /home/yoh/datalad/openfmri/ds000101 
2016-10-18 08:50:40,978 [DEBUG  ] Failed to detect a valid repo at /home/yoh/datalad/openfmri/ds000102 
...
```
I left this as is for now